src/main/java/homeWorkPart1/task6/BookRepository.java

```diff
@@ -0,0 +1,53 @@
+package homeWorkPart1.task6;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class BookRepository {
+
+    private List<Book> books = new ArrayList<>();
+
+    public void addBook(Book book) {
+        books.add(book);
+    }
+
+//    public void removeBook1(int id) {
+//        books.removeIf(book -> book.getId() == id);
+//    }
+
+    public void removeBook(int id) {
+        Iterator<Book> iterator = books.iterator();
```

Comment:
what about using `foreach` loop instead of iterator? also, using iterator might go wrong in this scenario because while you're iterating you're also removing, which might cause `ConcurrentModificationExcpetion` to be thrown. Search about this exception and how to prevent it